mllib/src/main/scala/org/apache/spark/ml/feature/Bucketizer.scala

```diff
@@ -128,8 +145,11 @@ object Bucketizer extends DefaultParamsReadable[Bucketizer] {
    * Binary searching in several buckets to place each data point.
    * @throws SparkException if a feature is < splits.head or > splits.last
    */
-  private[feature] def binarySearchForBuckets(splits: Array[Double], feature: Double): Double = {
-    if (feature.isNaN) {
+  private[feature] def binarySearchForBuckets(
+                                               splits: Array[Double],
+                                               feature: Double,
+                                               flag: String): Double = {
```

Comment:
Use a better name than "flag"  How about "keepNaNs"?
Also, please document this argument.